src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1276,6 +1276,12 @@ internal ValueTask<IBackendMessage> ReadMessage(
                 // We've read all the prepended response.
                 // Allow cancellation to proceed.
                 ReadingPrependedMessagesMRE.Set();
+
+                // User requested cancellation but it hasn't been performed yet.
+                // This might happen if the cancellation is requested while we're reading prepended responses
+                // because we shouldn't cancel them and otherwise might deadlock.
+                if (UserCancellationRequested && !PostgresCancellationPerformed)
+                    PerformDelayedUserCancellation();
             }
             catch (Exception e)
             {
```

Comment:
Should we want to remove https://github.com/npgsql/npgsql/blob/3fce77d7ce92b4ac984ce3afb4364b23b791fb05/src/Npgsql/Internal/NpgsqlConnector.cs#L2056 from Break? It doesn't seem to serve a purpose, as no method is actually waiting on the MRE.

Reply:
Unless I'm missing something it's already done?